### ISSUE TITLE
Songsteer download

### ISSUE THREAD

**Issue by thedoggaming16** [helge17/tuxguitar#102]
Yes hello, i think it's a bug but  if i download the Walk files in Songsterr (https://www.songsterr.com/a/wsa/pantera-walk-tab-s22271) and open it on Tuxguitar for i don't know what reason all of the tracks are like trumpets.... I don't know if it's a bug or anything but yea... it's weirdand now walk sound like taco bell anthem xD.

**Comment by thedoggaming16** [comment]
and =it also do this for Master of puppet, Fade to black, Symphony of destruction...

**Comment by guiv42** [comment]
@thedoggaming16: could you please elaborate a bit on the complete procedure, especially how you downloaded the tab?
I never used Songsterr, but when I look at the [FAQ](https://www.songsterr.com/a/wa/help#can-i-download-a-tab-for-offline-use) I find this:
```
Can I download a tab for offline use?
Songsterr web app is an online-only service. Try our mobile apps instead.

```

So I must admit I don't understand this issue!

**Comment by thedoggaming16** [comment]
Ok so first click here: https://www.songsterr.com/a/wsa/pantera-walk-tab-s22271
After click on Revised on:16/06/2023
And the last contrubution titeld the old one sounds off have a button to download (it going to download .gp file)
Open it and you going to see that all of the track are trumpet (same thing for master of puppets , fade to black, symphonium of destruction but not all the tabs do this)

**Comment by guiv42** [comment]
I followed the link in your post: is it normal the track in Songsterr is named "Phil Anselmo - Trumpet"?
![102](https://github.com/helge17/tuxguitar/assets/129443524/8d012f24-356a-460b-b92a-c9466d988cc2)

**Comment by thedoggaming16** [comment]
oki thx !

**Comment by vincentT** [comment]
I reproduced the issue. I downloaded the trial of GP8 and the file downloaded on songsterr (You need to be logged in to be able to edit the file and then download it) and works like a charm. However on tuxguitar 1.5.6 and 1.6.3 some instruments are not correctly recognized.
I faced it originally with this song https://www.songsterr.com/a/wsa/bullet-for-my-valentine-suffocating-under-words-of-sorrow-bass-tab-s23763?open=editor The 3 first tracks are correctly recognized as Distorsion guitar with dropped C tuning. The 4th track, the drums, is recognized as Piano and the 5th track, a dropped C tuned electric bass played with mediator, is recognized as drums, which is a mess to recover while trying to reassign the right instruments as it will automatically transpose the track (side question, is there a way to change the tuning of an instrument without transposing the track ? maybe it's a feature idea)
I also tested it on the song mentioned in the issue and there is a problem with the drum track (2nd track) that is also seen as a Piano.

**Comment by guiv42** [comment]
@vincentT thanks for testing with GP. So I think issue can be re-opened.

Could you please provide the guitar pro file directly in this conversation, under .zip format? (I have no songsterr account and don't want to create one).
If there's a copyright issue, you can empty the content of all tracks with GP8. I only need the list of tracks and their attributes.

**Comment by paulikauro** [comment]
Attached a file which seems to have the same issue: drums are recognized as piano and bass as drums, rest are fine. Additionally, all markers are missing after importing (there is only one marker in the attached file, but in the original all of them go missing).
The file is an emptied version of a tab. It was downloaded from (and hence generated by) Songsterr and has not been modified after downloading (apart from renaming the extension `gp` -> `zip`).

I don't have GuitarPro to test importing with that. I did test importing in MuseScore 4 and that worked (instruments and markers). Uploading back to Songsterr also works. I'm using TuxGuitar 1.6.3.

[test_with_marker.zip](https://github.com/user-attachments/files/16701140/test_with_marker.zip)

**Comment by guiv42** [comment]
Thanks for the file.
Opening this file with MuseScore 4.3.2:
- tracks 1,2,3: electric guitar, OK
- track 4: Cavaquinho, **KO**, should be acoustic guitar. Known [musescore issue](https://github.com/musescore/MuseScore/issues/21742)
- track 5: bass, OK
- track 6: drumkit, OK
- track 7: strings, I think it's **unexpected**, considering track name "synthesizer"

With TuxGuitar 1.6.3:
- tracks 1,2,3: electric guitar, OK
- track 4: acoustic guitar, OK
- track 5: drumkit, **KO**
- track 6: piano, **KO**
- track 7: strings, **unexpected** but consistent with musescore

the most critical problem is the confusion between percussion/non-percussion tracks (tracks 5 & 6), which is not easy to correct manually. I'll try to have a look.

**Comment by guiv42** [comment]
Let me reconsider this issue as "enhancement", and not "bug": the way songsterr fills the different fields of the gp file is clearly questionable.
TuxGuitar recognizes instruments thanks to the following xml nodes associated to each track:
- `<Sounds><Sound><MIDI><Program>`
- `<MidiConnection><PrimaryChannel>`
- `<MidiConnection><SecondaryChannel>`

Primary and secondary channels are left empty in the songsterr gp file. As far as I can see in the few .gp files I have, such fields are correctly set by GuitarPro, which sets them to 9 for percussion tracks. This is what TuxGuitar expects. This explains why track 5 is incorrectly identified as percussion: in absence of information, channels 1 to 8 are arbitrarily allocated to tracks 1 to 4, so channel 9 is allocated to track 5.
In 6th track, besides the undefined channels, midi program is set to zero, which corresponds to "Acoustic Grand Piano". No wonder it is recognized as such by TuxGuitar.

In other words: difficult to guarantee interoperability between songsterr and TuxGuitar relying on a proprietary and not well-defined file format.

**Comment by guiv42** [comment]
@paulikauro : about the marker issue, could you please tell me what is the text associated to the missing marker?

**Comment by paulikauro** [comment]
Thanks for looking into this!
Track 7 is indeed confusingly named, sorry about that. So the instrument being strings and track name synthesizer should be correct.
And yep, the marker's text is "Test".

**Comment by guiv42** [comment]
OK, thanks for your feedback.
I think I can fix that, but it will not be present in next release 1.6.4 (beta1 is already out it's a bit late). Will probably be in following release.

**Comment by guiv42** [comment]
That should be ok now (since release 2024-09-09)
@paulikauro, @vincentT: could you please test, and close issue if it's OK?

**Comment by vincentT** [comment]
I tested it on master branch and it works fine on my case, thanks 
However I'm not sure to have the right to close issues

**Comment by guiv42** [comment]
Thanks!
